### PR TITLE
feat: add env var to dump catalog responses

### PR DESCRIPTION
**Pull request by mkenigs**
When _FLOX_CATALOG_DUMP_RESPONSE_FILE is set, serialize the data a CatalogClient method is about to return.

This will be useful for debugging and generating test data.

### CODE REVIEW

**Review by dcarley — APPROVED** [review]
LGTM
```
% FLOX_FEATURES_USE_CATALOG=true _FLOX_CATALOG_DUMP_RESPONSE_FILE=catalog.json flox search hello >/dev/null 2>&1 && jq .count catalog.json
16
```

File: cli/flox-rust-sdk/src/providers/catalog.rs [hunk 1]
```diff
@@ -111,6 +112,21 @@ impl CatalogClient {
...
+        T: ?Sized + Serialize,
+    {
+        if let Ok(path_str) = std::env::var(FLOX_CATALOG_DUMP_DATA_VAR) {
+            std::fs::write(
+                PathBuf::from(path_str),
+                serde_json::to_string_pretty(response).unwrap(),
+            )
+            .unwrap();
+        }
+    }
```
Comment (ysndr) [inline cli/flox-rust-sdk/src/providers/catalog.rs:129]: i guess panicking is "ok" if this is primarily used internally but i think we should still use expect and some logging to figure out errors more easily.
(ignoring errors and logging them might actually be even safer)

```suggestion
        let Ok(path_str) = std::env::var(FLOX_CATALOG_DUMP_DATA_VAR) else {
            return;
        };

        tracing::debug!("writing response to {path_str}");
        let response_json =
            serde_json::to_string_pretty(response).expect("failed ot serialize response");

        std::fs::write(&path_str, response_json).expect("failed to write resopnse to dump file")
```

**Comment by ysndr** [comment]
If a command makes more than one request, all but the last request are being overridden.
Currently the only case where we make multiple requests are searches for suggestions after install attempts.
If this is going to be used for debugging, only dumping partial responses seems suboptimal

(some static atomic counter in the method can be used to provide unique sequential names for individual responses, or instead of overriding we can append to the dump file)

**Comment by billlevine** [comment]
> If a command makes more than one request, all but the last request are being overridden.
> Currently the only case where we make multiple requests are searches for suggestions after install attempts.
> If this is going to be used for debugging, only dumping partial responses seems suboptimal

I second this.  I think very quickly we'll want something like this... Consider a failing test suite that's multiple commands, or even 'in the field' diagnostics perhaps?  Can we append to the file? or read it in and append to an array?  The later would allow easy post processing with `jq` for test extraction etc.  Not sure how easy that is to do in rust.

That said, this is non-blocking and we can do this once it's actually needed.

**Review by billlevine — APPROVED** [review]

**Comment by dcarley** [comment]
> Can we append to the file? or read it in and append to an array? The later would allow easy post processing with `jq` for test extraction etc. Not sure how easy that is to do in rust.

It may not be pretty but `jq` will handle multiple documents in the same file if we append without wrapping in an array:
```
% (cat catalog.json catalog.json) | jq .count         
16
16
```

**Review by ysndr — APPROVED** [review]

**Pull request merged**